docs/proposals/20231122-automate-aks-features.md

```diff
@@ -0,0 +1,311 @@
+- [Alternatives](#alternatives)
+- [Upgrade Strategy](#upgrade-strategy)
+- [Additional Details](#additional-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Glossary
+
+## Summary
+
+CAPZ's AzureManagedControlPlane and AzureManagedMachinePool resources expose AKS's managed cluster and agent
+pool resources for Cluster API. Currently, new features in AKS require manual changes to teach CAPZ about
+those features for users to be able to take advantage of them natively in Cluster API. As a result, there are
+several AKS features available that cannot be used from Cluster API. This proposal describes how CAPZ will
+automatically make all AKS features available on an ongoing basis with minimal maintenance.
+
+## Motivation
+
+Historically, CAPZ has exposed an opinionated subset of AKS features that are tested and known to work within
+the Cluster API ecosystem. Since then, it has become increasingly clear that new AKS features are generally
+suitable to implement in CAPZ and users are interested in having all AKS features available to them from
+Cluster API.
+
+When gaps exist in the set of features available in AKS and what CAPZ offers, users of other infrastructure
+management solutions may not be able to adopt CAPZ. If all AKS features could be used from CAPZ, this would
+not be an issue.
+
+The AKS feature set changes rapidly alongside CAPZ's users' desire to utilize those features. Because making
+new AKS features available in CAPZ requires a considerable amount of mechanical, manual effort to implement,
+review, and test, requests for new AKS features account for a large portion of the cost to maintain CAPZ.
+
+Another long-standing feature request in CAPZ is the ability to adopt existing AKS clusters into management by
+Cluster API (https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1173). Making the entire AKS
+API surface area available from CAPZ is required to enable this so existing clusters' full representations can
+be reflected in Cluster API.
+
+### Goals
+
+- Narrow the gap between the sets of features offered by AKS and CAPZ.
+- Reduce the maintenance cost of making new AKS features available from CAPZ.
+- Preserve the behavior of existing CAPZ AKS definitions while allowing users to utilize the new API pattern
+  iteratively to use new features not currently implemented in CAPZ.
+
+### Non-Goals/Future Work
+
+- Automate the features available from other Azure services besides AKS.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a managed cluster user, I want to be able to use all available AKS features natively from CAPZ so that I
+can more consistently manage my CAPZ AKS clusters that use advanced or niche features without having to wait
```

Comment:
This sort of speaks to my comment below. Because there are several layers out of our control in terms of meeting AKS releases day-and-date, I think we want to re-word this user story to reflect something like this:

"As an AKS customer who uses CAPZ to manage clusters, I want to reduce the amount of wait time for new AKS features to be available for management by CAPZ."